src/Application/Actions/Person/Update.php

```diff
@@ -0,0 +1,149 @@
+        private readonly StripeClient $stripeClient,
+        private readonly ValidatorInterface $validator,
+    ) {
+        parent::__construct($logger);
+    }
+
+    /**
+     * @return Response
+     * @throws HttpBadRequestException
+     * @throws HttpNotFoundException
+     */
+    protected function action(): Response
+    {
+        $person = $this->personRepository->find($this->request->getAttribute('personId'));
+        if (!$person) {
+            throw new HttpNotFoundException($this->request, 'Person not found');
+        }
+
+        try {
+            /** @var Person $person */
+            $person = $this->serializer->deserialize(
+                $body = ((string) $this->request->getBody()),
+                Person::class,
+                'json'
+            );
+        } catch (UnexpectedValueException | TypeError $exception) {
+            // UnexpectedValueException is the Serializer one, not the global one
+            $this->logger->info(sprintf('%s non-serialisable payload was: %s', __CLASS__, $body));
+
+            $message = 'Person Update data deserialise error';
+            $exceptionType = get_class($exception);
+
+            return $this->validationError(
+                "$message: $exceptionType - {$exception->getMessage()}",
+                $message,
+                empty($body), // Suspected bot / junk traffic sometimes sends blank payload.
+            );
+        }
+
+        $violations = $this->validator->validate($person, null, ['complete']);
+
+        if (count($violations) > 0) {
+            $message = 'Validation error: ';
+
+            $violationDetails = [];
+            foreach ($violations as $violation) {
+                $violationDetails[] = $this->summariseConstraintViolation($violation);
+            }
+
+            $message .= implode('; ', $violationDetails);
+
+            return $this->validationError(
+                $message,
+                null,
+                true,
+            );
+        }
+
+        $person = $this->personRepository->persist($person);
+
+        $customerDetails = [
+            'email' => $person->email_address,
+            'name' => sprintf('%s %s', $person->first_name, $person->last_name),
+        ];
+
+        // Billing address can vary per payment method and is best kept against that object as it's
+        // the only thing we know the address matches.
+        // "Home address" is collected only for Gift Aid declarations and is optional, so append it conditionally.
+        if (!empty($person->home_address_line_1)) {
+            $customerDetails['address'] = [
+                'line1' => $person->home_address_line_1,
+            ];
+
+            if (!empty($person->home_postcode)) {
+                $customerDetails['address']['postal_code'] = $person->home_postcode;
+
+                // Should be 'GB' when postcode non-null.
+                $customerDetails['address']['country'] = $person->home_country_code;
```

Comment:
@NoelLH same as above regarding checking that `home_country_code` is not null

Reply:
As above. Also see the new field inline doc block for more detail.